inflation/sdp/InflationSDP.py

```diff
@@ -411,10 +409,26 @@ def set_bounds(self,
         """
         assert bound_type in ["up", "lo"], \
             "The 'bound_type' argument should be either 'up' or 'lo'"
+        if bounds is None:
+            return
+        # Sanitize list of bounds
+        sanitized_bounds = dict()
+        for mon, bound in bounds.items():
+            mon = self._sanitise_monomial(mon)
+            if mon not in sanitized_bounds.keys():
+                sanitized_bounds[mon] = bound
+            else:
+                old_bound = sanitized_bounds[mon]
+                assert np.isclose(old_bound, bound), \
+                    (f"Contradiction: Cannot set the same monomial {mon} to " +
+                     "have different upper bounds.")
         if bound_type == "up":
-            self._set_upperbounds(bounds)
+            self._reset_upperbounds()
+            self.moment_upperbounds = sanitized_bounds
         else:
-            self._set_lowerbounds(bounds)
+            self._reset_lowerbounds()
+            self.moment_lowerbounds.update(sanitized_bounds)
```

Comment:
Here I'm thinking about the case where a user does NOT want the 0 lower bounds on the physical monomials. (maybe they want to compare with standard NPA). If I understand correctly, if we just update `self.moment_lowerbounds` when using `set_bounds`, as opposed to setting the attribute, the user would have no way to remove previous bounds we are including by default. However, if we make this line not an update, but an assignment, then one could do it like this:

`sdp.set_bounds({**sdp.moment_lowerbounds, **{"pAB(00|00)": 0.2}}, "lo")` <-- this includes the physical bounds

`sdp.set_bounds({"pAB(00|00)": 0.2}}, "lo")` <-- this doesn't have the physical bounds

Any thoughts on this @apozas ?

Reply:
As described in the description of the PR, there exists a procedure for this, namely using `_update_xxxerbounds()` (intended for pro users).